init/fluent_bit_init_process.go

```diff
@@ -0,0 +1,401 @@
+	}
+
+	response, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		logrus.Fatalf("[FluentBit Init Process] Failed to read HTTP response: %s\n", err)
+	}
+	res.Body.Close()
+
+	err = json.Unmarshal(response, &metadata)
+	if err != nil {
+		logrus.Fatalf("[FluentBit Init Process] Failed to unmarshal ECS metadata: %s\n", err)
+	}
+
+	arn, err := arn.Parse(metadata.ECS_TASK_ARN)
+	if err != nil {
+		logrus.Fatalf("[FluentBit Init Process] Failed to parse ECS TaskARN: %s\n", err)
+	}
+
+	resourceID := strings.Split(arn.Resource, "/")
+	taskID := resourceID[len(resourceID)-1]
+	metadata.ECS_TASK_ID = taskID
+	metadata.AWS_REGION = arn.Region
+	metadata.ECS_TASK_DEFINITION = metadata.ECS_FAMILY + ":" + metadata.ECS_REVISION
+
+	return metadata
+}
+
+// set ECS Task Metadata as environment variables in the invoke_fluent_bit.sh
+func setECSTaskMetadata(metadata ECSTaskMetadata, filePath string) {
+
+	t := reflect.TypeOf(metadata)
+	v := reflect.ValueOf(metadata)
+
+	invokeFile := openFile(filePath)
+	defer invokeFile.Close()
+
+	for i := 0; i < t.NumField(); i++ {
+		if v.Field(i).Interface().(string) == "" {
+			continue
+		}
+		writeContent := "export " + t.Field(i).Name + "=" + v.Field(i).Interface().(string) + "\n"
+		_, err := invokeFile.WriteString(writeContent)
+		if err != nil {
+			logrus.Errorln(err)
+			logrus.Fatalf("[FluentBit Init Process] Cannot write %s in the invoke_fluent_bit.sh\n", writeContent[:len(writeContent)-2])
+		}
+	}
+}
+
+// create Fluent Bit command to use "-c" to specify the new main config file
+func createCommand(command *string, filePath string) {
+	*command = *command + " -c " + filePath
+}
+
+// create a folder to store S3 config files user specified (download them from S3)
+func createS3ConfigFileFolder(folderPath string) {
+	os.Mkdir(folderPath, 0700)
+}
+
+// get our built in config files or files from s3
+// process built-in config files directly
+// add S3 config files to folder "/init/fluent-bit-init-s3-files/"
+func getAllConfigFiles() {
+
+	// get all env vars in the container
+	envs := os.Environ()
+
+	// find all env vars match specified prefix
+	for _, env := range envs {
+		var envKey string
+		var envValue string
+		env_kv := strings.SplitN(env, "=", 2)
+		if len(env_kv) != 2 {
+			logrus.Fatalf("[FluentBit Init Process] Unrecognizable environment variables: %s\n", env)
+		}
+
+		envKey = string(env_kv[0])
+		envValue = string(env_kv[1])
+
+		matched_s3, _ := regexp.MatchString("aws_fluent_bit_init_s3", envKey)
+		matched_S3, _ := regexp.MatchString("aws_fluent_bit_init_S3", envKey)
+		matched_file, _ := regexp.MatchString("aws_fluent_bit_init_file", envKey)
+
+		// if this env var's value is an arn
+		if matched_s3 || matched_S3 {
+			getS3ConfigFile(envValue)
+		}
+		// if this env var's value is a path of our built-in config file, process is derectly
+		if matched_file {
+			processConfigFile(envValue)
+		}
+	}
+}
+
+func processConfigFile(path string) {
+
+	content_b, err := ioutil.ReadFile(path)
```

Comment:
what does `content_b` mean? Avoid to use meaningless name

Reply:
I think its always better to do full names if needed: `contentBytes`